File: 10 - Python/Exercícios/04 - Vetores/Exercício 12.py
#Exercício - Comerciante

N = int(input("Serão digitados dados de quantos produtos? "))

nome = [0] * N
precoCompra = [0] * N
precoVenda = [0] * N

totalCompra = 0
totalVenda = 0

for i in range(N):

    print(f"Produto {i+1}:")
    nome[i] = input("Nome: ")
    precoCompra[i] = float(input("Preço de compra: "))
    precoVenda[i] = float(input("Preço de venda: "))
    print("")

    totalCompra = totalCompra + precoCompra[i]
    totalVenda = totalVenda + precoVenda[i]

abaixo = 0
entre = 0
acima = 0

for i in range(N):

    lucro = precoVenda[i] - precoCompra[i]
    porc = lucro * 100 / precoCompra[i]

    if porc < 10:
        abaixo = abaixo + 1
    elif porc <= 20:
        entre = entre + 1
    else:
        acima = acima + 1

totalLucro = totalVenda - totalCompra

print("Relatório:")
print("Lucro abaixo de 10%:", abaixo)
print("Lucro entre 10% e 20%:", entre)
print("Lucro acima de 20%:", acima)
print("Valor total de compra:", round(totalCompra,2))
print("Valor total de venda:", round(totalVenda,2))
print("Lucro total:", round(totalLucro,2))

File: 06 - C/Exercícios/04 - Vetores/Exercício 12.c
//Exercício - Comerciante

#include <stdio.h>
#include <string.h>

int main(){

    int i, N, abaixo, entre, acima;
    double porc, lucro, totalCompra, totalVenda, totalLucro;

    printf("Serao digitados dados de quantos produtos? ");
    scanf("%d", &N);

    char nome[N][50];
    double precoCompra[N], precoVenda[N];

    totalCompra = 0;
    totalVenda = 0;

    for(i = 0; i < N; i++){

        printf("\nProduto %d:\n", i+1);
        printf("Nome: ");
		scanf("%s", &nome[i][0]);
        printf("Preco de compra: ");
        scanf("%lf", &precoCompra[i]);
        printf("Preco de venda: ");
        scanf("%lf", &precoVenda[i]);

        totalCompra = totalCompra + precoCompra[i];
        totalVenda = totalVenda + precoVenda[i];
    }

    abaixo = 0;
    entre = 0;
    acima = 0;

    for(i = 0; i < N; i++){

        lucro = (double) precoVenda[i] - precoCompra[i];
        porc = lucro * 100.0 / precoCompra[i];

        if(porc < 10.0){
            abaixo++;
        }
        else if(porc < 20){
            entre++;
        }
        else{
            acima++;
        }
    }

    totalLucro = totalVenda - totalCompra;

    printf("\nRELATORIO:\n");
	printf("Lucro abaixo de 10%%: %d\n", abaixo);
	printf("Lucro entre 10%% e 20%%: %d\n", entre);
	printf("Lucro acima de 20%%: %d\n", acima);
    printf("Valor total de compra: %.2lf\n", totalCompra);
    printf("Valor total de venda: %.2lf\n", totalVenda);
    printf("Lucro total: %.2lf\n", totalLucro);
    
    return 0;

}